rmw_opensplice_cpp/src/rmw_node.cpp

```diff
@@ -133,7 +133,7 @@ rmw_create_node(
   DDS::DomainParticipantQos dpqos;
   size_t length = strlen(name) + strlen("name=;") + 1;
   dp_factory->get_default_participant_qos(dpqos);
-  dpqos.user_data.value.length(length);
+  dpqos.user_data.value.length((DDS::ULong)length);
```

Comment:
Please avoid C casts in C++ code and use `static_cast<DDS::ULong>(length)` instead.